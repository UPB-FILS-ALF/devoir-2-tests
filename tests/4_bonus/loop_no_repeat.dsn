& LOOP no REPEAT

SET: %i 2
SET: %x 100
SET: %y 200
LOOP: 2
CIRCLE: 50 100 100 
WHILE: %i
LINE:  %x %y location
ADD: %x 100
SUB: %y 200
SUB: %i 1
REPEAT
