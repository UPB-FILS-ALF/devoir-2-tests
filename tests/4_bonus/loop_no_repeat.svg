<?xml version="1.0" encoding="UTF-8" ?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="1024" height="768">
	<!-- i = 2 -->
	<!-- x = 100 -->
	<!-- y = 200 -->
	<!-- loop 2: 2 -->
	<!-- loop #0 -->
		<circle cx="50" cy="100" r="100" stroke="rgb(0,0,0)"  stroke-width="1" fill="rgb(255,255,255)"/>
		<!-- while i: 2 -->
		<!-- while #0 i: 2 -->
			<line x1="0" y1="0" x2="100" y2="200" stroke="rgb(0,0,0)"  stroke-width="1" />
			<!-- x = x + 100 : 200 -->
			<!-- y = y - 200 : 0 -->
			<!-- i = i - 1 : 1 -->
			<!-- repeat -->
		<!-- while #1 i: 1 -->
			<line x1="100" y1="200" x2="200" y2="0" stroke="rgb(0,0,0)"  stroke-width="1" />
			<!-- x = x + 100 : 300 -->
			<!-- y = y - 200 : -200 -->
			<!-- i = i - 1 : 0 -->
			<!-- repeat -->
</svg>
