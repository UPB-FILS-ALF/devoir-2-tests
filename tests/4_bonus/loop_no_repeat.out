ERROR LINE (6): LOOP with no REPEAT
